### PR TITLE
refactor: split typstyle and test crates and organize test code

### PR DESCRIPTION
**Pull request by QuadnucYard**
Changes:

- Split sources into `typstyle-core` and `typstyle` crate. Move benches to the `typstyle-core` crate.
- Replace `once_cell::sync::Lazy` with `sync::LazyLock` (stable in 1.80).
- Move consistency checking with `reflexo` to `typstyle-consistency` crate. A `consistency` feature (enabled by default) controls this part. Now tests are unaware of typst compilation-related dependencies.
- Merge two test targets into one. This can avoid duplicated binary.
- Add bench to PR CI, although we have to check it manually.

### CODE REVIEW

File: crates/typstyle-consistency/Cargo.toml
Comment (Enter-tainer) [inline crates/typstyle-consistency/Cargo.toml:1]: consider adding publish=false in this crate. see https://doc.rust-lang.org/cargo/reference/manifest.html#the-publish-field

File: README.md [hunk 1]
```diff
@@ -1,19 +1,33 @@
 # typstyle
 
+A beautiful and reliable code formatter for Typst files.
```
Comment (Enter-tainer) [inline README.md:3]:
```suggestion
A beautiful and reliable code formatter for Typst.
```

File: Cargo.toml [hunk 2]
```diff
@@ -1,106 +1,64 @@
-[package]
-name = "typstyle"
+[workspace.package]
 version = "0.12.10"
+rust-version = "1.80"
```
Comment (Enter-tainer) [inline Cargo.toml:3]: is this msrv? Or are we locking rust version explicitly?

File: Cargo.toml [hunk 3]
```diff
@@ -1,106 +1,64 @@
...
-  "/LICENSE",
-  "/README.md",
-  "build.rs",
-  "tests/assets.rs",
-  "tests/repo-e2e.rs",
-]
+[workspace]
+resolver = "2"
+members = ["crates/*", "tests"]
+default-members = ["crates/typstyle"]
```
Comment (Enter-tainer) [inline Cargo.toml:14]: what will this field do?

File: .github/workflows/release.yml
Comment (Enter-tainer) [inline .github/workflows/release.yml:1]: we might need check if npm and crateio publish script still works. Let me do that

**Review by Enter-tainer — APPROVED** [review]
LGTM

File: Cargo.toml (hunk 2 above)
Comment (QuadnucYard) [reply Cargo.toml:3]: idk, I just take that of typst. Omit this field as before? Or use a newer version?

**Pull request merged**